.github/workflows/codeql.yml

```diff
@@ -0,0 +1,77 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ "main" ]
+  schedule:
+    - cron: '43 3 * * 6'
```

Comment:
is there a specific reason why you are running this at 3:43 ? 
just wondering. something like 00 0 * * 6 might have been better right ?

Reply:
No, you can schedule the workflow to run at any specific time you want. 
Ok, I am updating it to 00 0 *  * 6